### PR TITLE
make chained validation targeted

### PR DESCRIPTION
**Pull request by davidrichardson**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/validator/coordinator/ChainedValidationService.java [hunk 1]
```diff
@@ -1,53 +1,122 @@
...
+                            .findBySubmissionIdAndReference(submissionId, ref);
+                    return itemsReferencingTriggerSubmittable.stream();
+                })
+                .forEach(storedSubmittable -> revalidate(storedSubmittable,submissionId));
     }
 
-    protected Stream<? extends StoredSubmittable> streamSubmittablesInSubmissionExceptTriggerSubmittable(Submittable triggerSubmittable, String submissionId) {
-        Stream<? extends StoredSubmittable> streamSubmittablesInSubmission = streamSubmittablesInSubmission(submissionId);
+    protected AbstractSubsRef submittableToRef(Submittable submittable) {
+        AbstractSubsRef ref = new SampleGroupRef();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/coordinator/ChainedValidationService.java]: Do we need to instantiate the `ref` here with a concrete type?
Most of the type in the following lines we would replace them with another concrete type.

File: src/main/java/uk/ac/ebi/subs/validator/coordinator/CoordinatorListener.java [hunk 2]
```diff
@@ -245,11 +247,7 @@ public void processFileDeletionRequest(FileDeletedMessage fileDeletedMessage) {
     public void processSubmittableDeletion(StoredSubmittableDeleteMessage storedSubmittableDeleteMessage) {
         String submissionID = storedSubmittableDeleteMessage.getSubmissionId();
 
-        if (!fileValidationRequestHandler.handleFilesWhenSubmittableChanged(submissionID)) {
-            logger.error("Error handling submittable deleted from submission (id: {})", submissionID);
-        } else {
-            logger.trace("Triggering chained validation from submission (id: {})", submissionID);
-            chainedValidationService.triggerChainedValidation(null, submissionID);
-        }
+        fileValidationRequestHandler.handleFilesWhenSubmittableChanged(submissionID);
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/coordinator/CoordinatorListener.java]: We don't need the conditional check, anymore?
Why?

Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/validator/coordinator/CoordinatorListener.java]: I don't see that we needed it in the first place. The boolean returned tells us whether or not all the files in the submission had validation results. That doesn't affect whether or not I want to revalidate the other things in the submission

File: src/main/java/uk/ac/ebi/subs/validator/coordinator/ChainedValidationService.java (hunk 1 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/validator/coordinator/ChainedValidationService.java]: I've pushed an update to correct this

**Pull request merged**